### ISSUE TITLE
Feature Request: adding expiration date config

### ISSUE THREAD

**Issue by ngima** [frw/react-native-ssl-public-key-pinning#263]
Similar to the Android network security config which provide option to put the expiration date in <pin-set>, which helps to prevent the network connectivity issues in the app which is not updated with newer ssl pinning configuration, when user has disabled to auto update. 

``` xml
<?xml version="1.0" encoding="utf-8"?>
<network-security-config>

    <domain-config>
        <domain includeSubdomains="true">*.example.com</domain>
        <pin-set expiration="2024-01-01">
            <pin digest="SHA-256">...</pin>
            <!-- backup pin -->
            <pin digest="SHA-256">...</pin>
        </pin-set>
    </domain-config>
</network-security-config>
```

> The date, in yyyy-MM-dd format, on which the pins expire, thus disabling pinning. If the attribute is not set, then the pins do not expire.
Expiration helps prevent connectivity issues in apps which do not get updates to their pin set, such as when the user disables app updates.
https://developer.android.com/privacy-and-security/security-config#pin-set

Would be handy cool if we can pass expiration date in `DomainOptions`
``` Diff
export type DomainOptions = {
  /**
   * Whether all subdomains of the specified domain should also be pinned.
   * @default false
   */
  includeSubdomains?: boolean;
  /**
   * An array of SSL pins, where each pin is the base64-encoded SHA-256 hash of a certificate's Subject Public Key Info.
   * Note that at least two pins are needed per domain on iOS.
   */
  publicKeyHashes: string[];

+  expiration?: Date | string
};
```

And usage
``` Diff
await initializeSslPinning({
            'example.com': {
+                expiration: '2024-01-01',
                includeSubdomains: true,
                publicKeyHashes: [
                    'CLOmM1/OXvSPjw5UOYbAf9GKOxImEp9hhku9W90fHMk=',
                    'hxqRlPTu1bMS/0DITB1SSu0vd4u/8l8TjPgfaAp63Gc=',
                    'Vfd95BwDeSQo+NUYxVEEIlvkOlWY2SalKK1lPhzOx78=',
                    'QXnt2YHvdHR3tJYmQIr0Paosp6t/nggsEGD4QJZ3Q0g=',
                    'mEflZT5enoR1FuXLgYYGqnVEoZvmf9c2bVBpiOjYQ0c=',

                ],
            },
        });
```

**Comment by ngima** [comment]
Workaround

``` Typescript
import moment from 'moment';
import { initializeSslPinning } from 'react-native-ssl-public-key-pinning';

const EXPIRY_DATE = moment('2024-01-01')

export default async function initSslPinning(): Promise<void> {

    if (moment().isBefore(EXPIRY_DATE))
        return await initializeSslPinning({
            'example.com': {
                includeSubdomains: true,
                publicKeyHashes: [
                    'CLOmM1/OXvSPjw5UOYbAf9GKOxImEp9hhku9W90fHMk=',
                    'hxqRlPTu1bMS/0DITB1SSu0vd4u/8l8TjPgfaAp63Gc=',
                    'Vfd95BwDeSQo+NUYxVEEIlvkOlWY2SalKK1lPhzOx78=',
                    'QXnt2YHvdHR3tJYmQIr0Paosp6t/nggsEGD4QJZ3Q0g=',
                    'mEflZT5enoR1FuXLgYYGqnVEoZvmf9c2bVBpiOjYQ0c=',

                ],
            },
        });
}


```

**Comment by frw** [comment]
This is now implemented in v1.2.1 onwards.
